src/main/java/de/dhbw/karlsruhe/top/down/parsing/validation/TopDownAcceptorValidation.java

```diff
@@ -0,0 +1,199 @@
+            return false;
+        }
+
+        if (this.checkReadInput(stackCurrStep, inputNextStep) && this.checkSideConditionsRead(stackCurrStep, stackNextStep, productionCurrStep)) {
+            valid = true;
+        }
+
+        return valid;
+    }
+
+    private boolean checkReadInput(String stackCurrStep, String inputNextStep) {
+        boolean success = false;
+
+        char firstCharStackCurrStep = stackCurrStep.charAt(0);
+        char firstCharInputNextStep = inputNextStep.charAt(inputNextStep.length()-1);
+
+        if (firstCharStackCurrStep == firstCharInputNextStep) {
+            success = true;
+        }
+
+        return success;
+    }
+
+    private boolean checkSideConditionsRead(String stackCurrStep, String stackNextStep, GrammarRule productionCurrStep) {
+        boolean success = false;
+
+        char secondCharStackCurrStep = stackCurrStep.charAt(1);
+        char firstCharStackNextStep = stackNextStep.charAt(0);
+
+        if (productionCurrStep == null && (secondCharStackCurrStep == firstCharStackNextStep)) {
+            success = true;
+        }
+
+        return success;
+    }
+
+    private boolean validateProductionStep(TopDownStep tdStep, TopDownStep nextStep) {
+        boolean valid = false;
+
+        String stackCurrStep = tdStep.getNewStack();
+        String stackNextStep = nextStep.getNewStack();
+        GrammarRule productionCurrStep = tdStep.getUsedProduction();
+        String inputCurrStep = tdStep.getReadInput();
+        String inputNextStep = nextStep.getReadInput();
+        TopDownState stateCurrStep = tdStep.getNewState();
+
+        if (stackCurrStep == null || productionCurrStep == null || stateCurrStep != TopDownState.Z) {
+            return false;
+        }
+
+        if (this.checkUsedProduction(stackCurrStep, stackNextStep, productionCurrStep) && this.checkSideConditionsProduction(inputCurrStep, inputNextStep)) {
+            valid = true;
+        }
+
+        return valid;
+    }
+
+    private boolean checkUsedProduction(String stackCurrStep, String stackNextStep, GrammarRule productionCurrStep) {
+        boolean success = false;
+
+        String leftSide = productionCurrStep.leftSide();
+        String rightSide = productionCurrStep.rightSide();
+
+        if (rightSide.equals("epsilon")) {
+            String stackCurrStepWOT = stackCurrStep.substring(1);
+
+            if (stackCurrStepWOT.equals(stackNextStep)) {
+                success = true;
+            }
+        } else {
+            String firstCharStackCurrStep = String.valueOf(stackCurrStep.charAt(0));
+            String firstCharsInputNextStep = stackNextStep.substring(0, rightSide.length());
+
+            if (firstCharStackCurrStep.equals(leftSide) && firstCharsInputNextStep.equals(rightSide)) {
+                success = true;
+            }
+        }
+
+        return success;
+    }
+
+    private boolean checkSideConditionsProduction(String stackCurrStep, String stackNextStep) {
```

Comment:
Siehe oben: Methodenname ist hier etwas unklar: Was sind die Side conditions und wieso sind sie nur "side" conditions? Vielleicht mit der Methode checkUsedProduction zusammenführen oder eindeutiger benennen. Wobei das hier auch so kurz ist, dass stackCurrStep.equals(stackNextStep) an den Stellen, an denen die Methode aufgerufen wird eingesetzt werden könnte.

Reply:
Die "side conditions" dienen zum verifizieren des restlichen Ablaufs. Dabei wird nicht der Lese/Produktionsschritt an sich überprüft, sondern ob die Zeichen in der nächsten Eingabe richtig angegeben und nichts vergessen zu entfernen/hinzufügen wurde.

Die Funktionalität ist etwas zu lang, um sie in den Methodennamen zu packen, weshalb ich mich dazu entschieden habe es in einem Kommentar kurz zu erläutern. Genauere Infos gibt es ja dann in der Ausarbeitung und in einer eventuellen Doc.